cmd/operator-sdk/olmcatalog/gen-csv.go

```diff
@@ -79,30 +80,46 @@ func genCSVFunc(cmd *cobra.Command, args []string) error {
 	if projutil.IsOperatorGo() {
 		cfg.Repo = projutil.CheckAndGetProjectGoPkg()
 	}
+	s := &scaffold.Scaffold{}
+
+	if cmd.Flags().Changed(writeCSVConfigOpt) {
+		if err := writeConfig(s, cfg); err != nil {
+			return err
+		}
+	}
+
+	var csvCfgFile *catalog.CSVConfigFile
+	if _, err = os.Stat(csvConfigPath); err == nil {
+		log.Infof("Using custom CSV config %s.", csvConfigPath)
+		csvCfgFile, err = catalog.GetCSVConfigFile(csvConfigPath)
+		if err != nil {
+			return err
+		}
+	} else if err != nil && csvConfigPath != "" {
+		return fmt.Errorf("error reading CSV config %s: %v", csvConfigPath, err)
+	} else {
+		log.Info("Using default CSV config.")
+		csvCfgFile = &catalog.CSVConfigFile{}
+	}
 
```

Comment:
could not it be improved? 
By default, the default CSVFile be set then if do not find it or the customized one return an error? 
Could not the logic be when the new struct is created?